src/lib/agent/PolykeyAgent.ts

```diff
@@ -402,33 +471,52 @@ class PolykeyAgent {
   private async listVaults(nodePath: string) {
     const pk = this.getPolyKey(nodePath);
     const vaultNames = pk.vaultManager.listVaults();
-    return ListVaultsResponseMessage.encode({ vaultNames }).finish();
+    return ListVaultsResponseMessage.encodeDelimited({ vaultNames }).finish();
+  }
+  private async scanVaultNames(nodePath: string, request: Uint8Array) {
+    const { publicKey } = ScanVaultNamesRequestMessage.decodeDelimited(request);
+    const pk = this.getPolyKey(nodePath);
+    const vaultNames = await pk.vaultManager.scanVaultNames(publicKey);
+    return ScanVaultNamesResponseMessage.encodeDelimited({ vaultNames }).finish();
   }
   private async newVault(nodePath: string, request: Uint8Array) {
-    const { vaultName } = NewVaultRequestMessage.decode(request);
+    const { vaultName } = NewVaultRequestMessage.decodeDelimited(request);
     const pk = this.getPolyKey(nodePath);
     await pk.vaultManager.createVault(vaultName);
-    return NewVaultResponseMessage.encode({ successful: true }).finish();
+    return NewVaultResponseMessage.encodeDelimited({ successful: true }).finish();
+  }
+  private async cloneVault(nodePath: string, request: Uint8Array) {
+    const { vaultName, publicKey } = CloneVaultRequestMessage.decodeDelimited(request);
+    const pk = this.getPolyKey(nodePath);
+    await pk.vaultManager.cloneVault(vaultName, publicKey);
+    return CloneVaultResponseMessage.encodeDelimited({ successful: true }).finish();
+  }
+  private async pullVault(nodePath: string, request: Uint8Array) {
```

Comment:
Can you add in eslint rules to ensure that we have a whitespace between class methods?

Reply:
sure

Reply:
done.